app/helpers/admin/audit_trail_helper.rb

```diff
@@ -19,7 +19,11 @@ def describe_audit_trail_entry(entry)
   end
 
   def paginated_audit_trail_url(page)
-    url_for(params.permit!.except(:host, :controller, :action).merge(controller: "admin/edition_audit_trail", action: "index", page: (page <= 1 ? nil : page)))
+    url_for(
+      params.permit(:id, :page, :_)
+            .except(:host, :controller, :action)
```

Comment:
If these aren't permitted I'm not sure why we'd need to except them? though if they might exist we'll get a warning so probably needs a slice.

I think you might be able to use `params.to_unsafe_hash` maybe? I can see stuff in [brakeman for handling those in redirects](https://github.com/presidentbeef/brakeman/pull/1029) so might not raise any warnings as part of URL generation.

What is the `:_` parameter?

Reply:
I've removed the except code and gone with slice and permit like suggested prior to this comment.

> What is the :_ parameter?

I have no idea, added it as its a parameter being passed around in the audit trail feature test when clicking between pages but I've looked for it in the UI and can't seem to find the source of the mysterious underscore param so have removed it from the slice!